### PR TITLE
Implement current state enum in client

### PR DESCRIPTION
**Pull request by AnastasiiaSvietlova**
## Changeset

Add a new `appState` property and `setAppState` to the core client.

## Testing

Add unit test to test `setAppState` 
Update `createTestClient` default options

### CODE REVIEW

**Review by gingerbenw — APPROVED** [review]
Looks good - but could we create a new `integration/component-rendering` branch and target that please? 🙏🏻

File: packages/test-utilities/lib/create-test-client.ts [hunk 1]
```diff
@@ -25,6 +26,8 @@ const defaultOptions = () => ({
   spanAttributesSource,
   schema,
   plugins: () => [],
+  appState: 'starting',
+  setAppState: () => AppState,
```
Comment (gingerbenw) [inline packages/test-utilities/lib/create-test-client.ts:30]:
```suggestion
  setAppState: (appState: AppState) => void,
```

**Review by gingerbenw — CHANGES_REQUESTED** [review]
Sorry, just spotted another change

File: packages/test-utilities/lib/create-test-client.ts (hunk 1 above)
Comment (AnastasiiaSvietlova) [reply packages/test-utilities/lib/create-test-client.ts:30]: In this case `'appState' is declared but its value is never read`
Maybe:  `setAppState: (appState: AppState) => {appState} `?

**Review by gingerbenw — APPROVED** [review]

**Pull request merged**